### PR TITLE
Efficient decoder text generation wrapper

### PR DESCRIPTION
**Pull request by jimypbr**
# What does this PR do?

Current text generation returns all the logits back from IPU to host. However we only need the logits from the token to be generated. This wraps the model to return only what is necessary to improve the IO performance of the text generation.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/graphcore/pipelines/__init__.py [hunk 1]
```diff
@@ -411,8 +411,10 @@ def new_forward(self, model_inputs, *args, **kwargs):
     # Implement pipelines __del__ to clean up poplar exector
     def _del(self):
         # For text generation models, deallocate the internal poplar executor
-        if hasattr(self.model, "poptorch_model"):
-            self.model.poptorch_model.destroy()
+        if hasattr(self.model, "poptorch_decoder"):
+            self.model.poptorch_decoder.destroy()
+        if hasattr(self.model, "poptorch_encoder"):
```
Comment (katalinic-gc) [inline optimum/graphcore/pipelines/__init__.py:416]: Is that the case yet?

**Review by katalinic-gc — APPROVED** [review]
lgtm
